### PR TITLE
Add optional hostname logging to log messages.

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/656
Docs PR: https://github.com/arangodb/docs/pull/628

Add optional hostname logging to log messages.

Whether or not the hostname is added to each log message can be controlled via
the new startup option `--log.hostname`. Its default value is the empty string,
meaning no hostname will be added to log messages.
Setting the option to an arbitrary string value will make this string be logged
in front of each regular log message, and inside the `hostname` attribute in
case of JSON-based logging. Setting the option to a value of `auto` will use
the hostname as returned by `gethostbyname`.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/628
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/656

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (e.g. in server_parameters)

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/14189/

### CODE REVIEW

File: lib/Logger/LoggerFeature.cpp [hunk 1]
```diff
@@ -199,10 +199,15 @@ void LoggerFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
...
+  options->addOption("--log.hostname",
+                     "hostname to use in log message (empty for none, use 'auto' to automatically figure out hostname)",
+                     new StringParameter(&_hostname))
+                     .setIntroducedIn(30800);
+  
   options->addOption("--log.process", "show process identifier (pid) in log message",
                      new BooleanParameter(&_processId),
                      arangodb::options::makeDefaultFlags(arangodb::options::Flags::Hidden))
-                     .setIntroducedIn(30701);
+                     .setIntroducedIn(30800);
```
Comment (joerg84) [inline lib/Logger/LoggerFeature.cpp:210]: Is this related/intended?

**Review by joerg84 — APPROVED** [review]
Except for one question, lgt

File: lib/Logger/Logger.cpp [hunk 2]
```diff
@@ -429,7 +445,7 @@ void Logger::log(char const* logid, char const* function, char const* file, int
     // tid
     if (_showThreadIdentifier) {
       out.append(",\"tid\":");
-      StringUtils::itoa(uint64_t(Thread::currentThreadNumber), out);
+      StringUtils::itoa(uint64_t(Thread::currentThreadNumber()), out);
```
Comment (Dronplane) [inline lib/Logger/Logger.cpp:448]: 🙈

File: lib/Logger/Logger.cpp [hunk 3]
```diff
@@ -528,10 +550,16 @@ void Logger::log(char const* logid, char const* function, char const* file, int
...
+    if (!_hostname.empty()) {
+      out.append(_hostname);
+      out.push_back(' ');
+    }
+
     // human readable format
     LogTimeFormats::writeTime(out, _timeFormat, std::chrono::system_clock::now());
     out.push_back(' ');
-
+
```
Comment (Dronplane) [inline lib/Logger/Logger.cpp:562]: Do we need spaces here? Looks like an accident formatting change

**Review by Dronplane — APPROVED** [review]
LGTM

File: lib/Logger/LoggerFeature.cpp (hunk 1 above)
Comment (jsteemann) [reply lib/Logger/LoggerFeature.cpp:210]: Yes, this is intended. I found this while adding the new option, and found that the already existing option `--log.process` was only added in 3.8, but not in 3.7.1.
So I fixed it.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/14203/
